### PR TITLE
Migrate spec to OpenAPI 3 and new swagger-repo

### PR DESCRIPTION
**Pull request by RomanGotsiy**
~Please, do not merge 💥~

~I have to review it carefully again in the morning~

~After that, I will list some limitations/breaking changes~

Can be merged!

### CODE REVIEW

**Comment by RomanGotsiy** [comment]
I reviewed it again and fixed a few fixes I found. Now looks ok to me.

Note, that this PR is just simple migration from OAS2 to OAS3. I didn't try to improve the spec by using new OAS3 features and you are welcome to do it incrementally. Some notably improvements in OAS3:
- `discriminator` [now supports `propertyName` and `mapping`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#discriminator-object), both are supported by ReDoc.
- `oneOf` and `anyOf` are available which you may find useful. Although, I would recommend you to use it as less as possible as this keyword is good for validation only and works bad for Docs/Code generation/etc.
- there is `examples` field on [`Media Type Object`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#fixed-fields-12) which you can use to specify more than one example for request body or response. Partially supported by ReDoc (no support for `externalValue`). Also, you can add `examples` folder to `components` folder and reference examples instead of referencing them directly.

Some breaking changes:
- Relative paths from the root of the document to specific definitions have changed. It has to be updated in your Front-End
- `_link` used to be documented as array which implied order. OpenAPI 3 disallowed this use case. So I replaced it with `anyOf` for now. I will implement some way to backport previous functionality.

@slavcodev @adamaltman Please, review.

File: spec/paths/email-messages@{id}.yaml [hunk 1]
```diff
@@ -11,71 +11,75 @@ get:
...
 delete:
   tags:
-  - Email Messages
+    - Email Messages
   summary: Delete an email message
   description: |
     Delete an email message
   responses:
-    204:
+    '204':
```
Comment (slavcodev) [inline spec/paths/email-messages@{id}.yaml]:
```suggestion
    204:
```

File: spec/paths/gateway-accounts@{id}@downtime-schedules@{downtimeId}.yaml [hunk 2]
```diff
@@ -17,69 +17,70 @@ get:
...
+    $ref: '#/components/requestBodies/GatewayAccountDowntimeSchedule'
 delete:
   tags:
     - Gateway Accounts
   summary: Delete a Gateway Account downtime schedule
   description: |
     Delete a Gateway Account downtime schedule
   responses:
-    204:
+    '204':
```
Comment (slavcodev) [inline spec/paths/gateway-accounts@{id}@downtime-schedules@{downtimeId}.yaml]:
```suggestion
    204:
```

File: spec/paths/gateway-accounts@{id}@timeline@{messageId}.yaml [hunk 3]
```diff
@@ -17,36 +17,38 @@ get:
...
 delete:
   tags:
-  - Gateway Accounts
+    - Gateway Accounts
   summary: Delete a Gateway Timeline message
   description: |
     Delete a Gateway Timeline message with predefined identifier string
   responses:
-    204:
+    '204':
```
Comment (slavcodev) [inline spec/paths/gateway-accounts@{id}@timeline@{messageId}.yaml]:
```suggestion
    204:
```

File: spec/paths/layouts@{id}.yaml [hunk 4]
```diff
@@ -11,76 +11,77 @@ get:
...
+      $ref: '#/components/responses/InvalidDataError'
 delete:
   tags:
     - Layouts
   summary: Delete a layout
   description: |
     Delete a layout with predefined identifier string
   responses:
-    204:
+    '204':
```
Comment (slavcodev) [inline spec/paths/layouts@{id}.yaml]:
```suggestion
    204:
```

File: spec/paths/lists@{id}.yaml [hunk 5]
```diff
@@ -11,78 +11,79 @@ get:
...
+      $ref: '#/components/responses/InvalidDataError'
 delete:
   tags:
     - Lists
   summary: Delete a list
   description: |
     Delete a list with predefined identifier string
   responses:
-    204:
+    '204':
```
Comment (slavcodev) [inline spec/paths/lists@{id}.yaml]:
```suggestion
    204:
```

**Comment by adamaltman** [comment]
Did you intend to remove `yarn.lock` @RomanGotsiy ?

**Comment by XaosSintez** [comment]
I really love `discriminator` changes 💯 
PR looks ok to me, although i didn't review it very carefully. Preview seems ok too.

File: spec/components/responses/SeeOther.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      type: integer
+  Rate-Limit-Remaining:
+    description: The number of remaining requests in the current period
+    schema:
+      type: integer
+  Rate-Limit-Reset:
+    description: >
+      The date in format defined by [RFC
+      822](https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.3.1)
+
```
Comment (ashkarpetin) [inline spec/components/responses/SeeOther.yaml]: Bad formatting

File: spec/components/parameters/collectionFilter.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+name: filter
+in: query
+description: >
+  The collection items filter requires a special format.
+
```
Comment (ashkarpetin) [inline spec/components/parameters/collectionFilter.yaml]: No need empty lines between, I think.

File: spec/components/securitySchemes/JWT.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+description: >
+  You can create a JSON Web Token (JWT) via our [JWT Session
+  resource](https://rebilly.github.io/RebillyUserAPI/#tag/JWT-Session/paths/~1signin/post).
+
```
Comment (ashkarpetin) [inline spec/components/securitySchemes/JWT.yaml]: No need to divide by empty line

File: spec/components/securitySchemes/PublishableApiKey.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+description: >
+  Only for the Tokens resource.
+
```
Comment (ashkarpetin) [inline spec/components/securitySchemes/PublishableApiKey.yaml]: No need to divide by empty line

**Comment by slavcodev** [comment]
> @slavcodev In OAS 3 examples '204' is used (https://swagger.io/docs/specification/describing-responses/). Though I'm not sure if we should follow this style.

@ashkarpetin ohh, nice. Seems I should reread the specs, I missed this change. Response code is string in `OAS3`, since it support ranges :)

**Comment by RomanGotsiy** [comment]
Fixed formatting. Thanks, @ashkarpetin.

I think it can be merged now.

**Comment by RomanGotsiy** [comment]
@adamaltman yes. It is not so good to keep both `yarn.lock` and `package-lock.json` at the same time.
Although I can bring `yarn.lock` back if most the people who work on this repo are using `yarn` vs `npm`

**Comment by adamaltman** [comment]
I think most people use yarn vs. npm here.  Also there is one conflict to resolve.  Otherwise, I think It would be good to merge it and on to RebillyAPI.

**Comment by RomanGotsiy** [comment]
`yarn.lock` is back + updated readme.

**Another breaking change**:
- `swagger.json`, `swagger.yaml` files in the output are now renamed to `openapi.json` and `openapi.yaml` respectively

**Pull request merged**